pkg/machines/libvirt-dbus.js

```diff
@@ -1123,4 +1127,9 @@ function updateNetworkIface({ domXml, networkMac, networkState, networkModelType
     return null;
 }
 
+export function storagePoolDelete(connectionName, objPath) {
+    return clientLibvirt[connectionName].call(objPath, 'org.libvirt.StoragePool', 'Delete', [Enum.VIR_STORAGE_POOL_DELETE_NORMAL], TIMEOUT)
+            .then(() => clientLibvirt[connectionName].call(objPath, 'org.libvirt.StoragePool', 'Undefine', [], TIMEOUT));
+}
```

Comment:
It might be good idea to give user an option to only udefine storage pool or delete it completely including the storage.  In addition not all storage pools supports delete operation, currently it's only DIR, FS, NFS, LOGICAL, VSTORAGE and ZFS.

Reply:
Why only undefine the storage pool without deletion? One doesn't have to option to re-define existing storage anyway. One can only create one from scratch.

Reply:
That is not true, cockpit is not calling `virStoragePoolBuild` libvirt API and when calling `virStoragePoolCreate` cockpit is not using any of the `build` flags so the storage pool is not created from scratch, it's just defining some existing path on the host.

So for example in case if DIR storage pool the delete operation removes the directory as well and build operation would create all necessary directories in order to use that storage pool.  The storage directory creation is not supported by cockpit, if you are creating new storage pool you need to provide existing directory therefore the `build` operation is not possible.

Counter-part to current storage creation in cockpit is only `undefine` from libvirt.  For that reason the `delete` option should be optional.

Reply:
Good point that are not calling virStoragePoolBuild API. This has to be fixed, we need to support building storage pools as well. virt-manager allows that and we should here as well.

On the other hand how would you imagine the UI to allow undefining pools without deleting them? virt-manager doesn't offer this option from what I see.
And what about the storage volumes they contain. Should we force the user to remove first all volumes and then attempt pools deletion of should be force delete?